lib/concurrent/configuration.rb

```diff
@@ -1,124 +1,296 @@
 require 'thread'
-require 'concurrent/delay'
+require 'concurrent/lazy_reference'
+require 'concurrent/atomics'
 require 'concurrent/errors'
-require 'concurrent/atomic'
-require 'concurrent/executor/immediate_executor'
-require 'concurrent/executor/thread_pool_executor'
-require 'concurrent/executor/timer_set'
+require 'concurrent/executors'
 require 'concurrent/utility/processor_count'
 
 module Concurrent
   extend Logging
 
-  # A gem-level configuration object.
-  class Configuration
+  # Suppresses all output when used for logging.
+  NULL_LOGGER = lambda { |level, progname, message = nil, &block| }
 
-    # a proc defining how to log messages, its interface has to be:
-    #   lambda { |level, progname, message = nil, &block| _ }
-    attr_accessor :logger
+  # initialize the global executors
+  class << self
```

Comment:
To get the appropriate context for the class variables which need to be access from within the class methods. This code runs when the file is first loaded by Ruby, this ensuring that all those objects get created when the library loads. If we didn't wrap this code in the `self << class` block the class variables would not have been the same as what is being referenced in the class methods.